axelrod/strategy_transformers.py

```diff
@@ -228,11 +297,21 @@ def dual_wrapper(player, opponent, proposed_action):
     -------
     action: an axelrod.Action, C or D
     """
+
     if not player.history:
-        player.original_player = player.original_class(**player.init_kwargs)
+        player.use_history = []
+
+    temp = player.history[:]
+    player.history = player.use_history[:]
+
```

Comment:
Here's my thinking process:

`op` = original_player
`wp` = wrapped_player

`play` only mutates history, cooperations, defections, state_distributions.

all other player attributes are mutated in `player.strategy`

`play_attrs` = history, cooperations, defections, state_distributions

`wp` strategy:

1) switch to `op.play_attrs`
2) play `op` strategy
3) this will update `wp` state as if it was `op`
4) update history with `op` action - this step could be skipped and done on-the-fly in step 1 (see code below)
5) switch back to `wp.play_attrs`
6) flip action
7) return (assuming play was called, `play` will now update `wp.play_attrs`.  this assumption was also implicit in the original implementation.)

the resulting player is a `wp` with `wp` history but `op` state.  

code to switch play_attrs (as yet untested.  This is just to get the gist):

```
def switch_play_attrs(player):
    new_history = [action.flip() for action in player.history]
    player.history = new_history
    
    temp = player.cooperations
    player.cooperations = player.defections
    player.defections = temp
 
    new_distributions = defaultdict(int)
    for key, val in player.state_distribution.items():
        new_key = (key[0].flip(), key[1])
        new_distributions[new_key] = val
    player.state_distribution = new_distributions
```